Credit contributor (#135)
OpenTermsArchive/contribution-tool: content/parts/en/contributor-form.mdx

## Sign your contribution

Open Terms Archive is an open source initiative that relies on contributors just like you 🤗

By default, your contributions will be anonymous. Signing your contributions enables you to be credited publicly for them and to be notified when they are integrated or discussed.

<ContributorFormFields />

> - While not readable anywhere on the web, this email address will be [publicly](https://stackoverflow.com/a/897604/594053) and irreversibly associated with your contribution.

Reviewer: > While not readable anywhere on the web

I don’t think that’s true? Or am I missing something?

The repository is hosted on GitHub, and GitHub displays the Git commit information like author and author email fields as well as the commit message.

Author: You're right on the fact that it can be read, but on the commit page through, I could not find any place mentioning the email.

![Screen Shot 2022-11-14 at 11 36 33](https://user-images.githubusercontent.com/4191809/201601572-830aaa3b-0bae-4b24-892d-e3cc98aad6ec.png)

It's available through an API call though.

<img width="725" alt="Screen Shot 2022-11-14 at 11 37 59" src="https://user-images.githubusercontent.com/4191809/201601846-2771aa88-577d-45fa-8a56-bbb15cf9e14b.png">

Reviewer 2: Indeed, thanks @martinratinaud for the examples 🙂 

Hence my phrasing: this is not available on the web, but is public. Anyone cloning the repo can find it. Not through a web search though, and this will not be indexed.

- We will use this address only if it is necessary to contact you for integrating, crediting or otherwise handling this specific contribution.
- If you have an account on [GitHub](https://github.com), using an address linked to it will enable mentioning your account.

<small>

>   If you only want to contribute anonymously, you can leave the default email. You can always join the discussion later, but maintainers will not be able to proactively contact you if they have questions.

Reviewer: I would drop the *only*

```suggestion
  If you want to contribute anonymously, you can leave the default email. You can always join the discussion later, but maintainers will not be able to proactively contact you if they have questions.
```

It simplifies the sentence. One may argue about the subjective superiority/implied value/nudging - but I think even without that the simpler sentence is worth it.

For us it does not really matter if a contribution is anonymous or credited.

>   If you only want to contribute anonymously, you can leave the default email. You can always join the discussion later, but maintainers will not be able to proactively contact you if they have questions.

Reviewer: > You can always join the discussion later, but maintainers will not be able to proactively contact you if they have questions.

I assume this is mainly referring to joining a PR discussion, and maintainers contacting related to the PR before merging?

With the email use disclosure also listing “otherwise handling this specific contribution” I was also thinking of if there is relicensing and other copyright related requests/communication. Is that also considered a possibility, and if not, what does other handling refer to/enclose?

## Sign your contribution

> Open Terms Archive is an open source initiative that relies on contributors just like you 🤗.

Reviewer 2: ```suggestion
Open Terms Archive is an open source initiative that relies on contributors just like you 🤗
```

> By default, your contributions will be anonymous. Signing your contributions with your name and/or email address will enable you to be notified when they are integrated or discussed and to be credited publicly for them.

Reviewer 2: ```suggestion
By default, your contributions will be anonymous. Signing your contributions will enable you to be notified when they are integrated or discussed and to be credited publicly for them.
```

<SignatureNameInput />
<SignatureEmailInput />

<UpdateButton>Save your signature</UpdateButton>

- While not readable anywhere on the web, this email address will be [publicly](https://stackoverflow.com/a/897604/594053) and irreversibly associated with your contribution.
- We will use this address only if it is necessary to contact you for integrating, crediting or otherwise handling this specific contribution.
- If you have an account on [GitHub](https://github.com), we recommend using an address linked to it so your account is mentioned.

<small>
  If you only want to contribute anonymously, you can leave the default email. You can always join

>   the discussion later, but we will not be able to proactively contact you if we have questions.

Reviewer 2: ```suggestion
  the discussion later, but maintainers will not be able to proactively contact you if they have questions.
```

> ## Get credit for your contribution!
> 
> Open Terms Archive is an Open Source initiative that relies on contributors just like you.
> 
> We thus want to give you **full credit** for your contribution and in order to do that we need some small information from you.
> 
> 2 options now:
> 
> ### You want to take credit
> 
> Awesome! Please, enter either of the following in the field below:
> 
> - Your GitHub email address → You will be <a href="https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors" target="_blank">listed as a co-author</a>
> - Any email address → This address will be visible on the change proposed on GitHub
> 
> <Input />
> <UpdateButton>Save email</UpdateButton>
> 
> ---
> 
> ### You don't want to take credit
> 
> That's fine, just <IgnoreButton>click here</IgnoreButton>.

Reviewer 2: ```suggestion
## Sign your contribution

Open Terms Archive is an open source initiative that relies on contributors just like you 🤗 By default, your contributions will be anonymous. Signing your contributions with your email address will enable you to be notified when they are integrated or discussed and to be credited publicly for them.

<Input />
<UpdateButton>Sign your contribution with your email address</UpdateButton>

- While not readable anywhere on the web, this email address will be [publicly](https://stackoverflow.com/a/897604/594053) and irreversibly associated with your contribution.
- We will use this address only if it is necessary to contact you for integrating, crediting or otherwise handling this specific contribution.
- If you have an account on [GitHub](https://github.com), we recommend using an address linked to it so your account is mentioned.

<small>If you only want to contribute anonymously, you can <IgnoreButton>opt-out now</IgnoreButton>. You can always join the discussion later, but we will not be able to proactively contact you if we have questions.</small>
```
